### PR TITLE
Adding UI for Updates and Migration Information

### LINKED ISSUE #52265: Add UI to migrator
We need a way to empower users to manage and fix there own database drift when doing migrations. We need to create a UI for system admins to manage this process.

### PR DESCRIPTION
**Pull request by Numbers88s**
**Description**

This pull request introduces a user interface (UI) for updates and migration information in the migrator. The UI provides users with comprehensive information regarding their migration status and detects any potential drifts in the system. It also offers helpful suggestions for resolving any identified drifts.

**Changes Made**

- Added a new UI component for displaying update and migration information.
- Implemented functionality to detect drifts and display relevant details.
- Integrated the ability to generate deltas for identified drifts.
- Incorporated support for providing users with executable statements to address drifts.

**New Features**

- The UI presents users with an overview of their migration status.
- It highlights any detected drifts and provides detailed information for each.
- The UI offers suggestions for addressing drifts and provides instructions for running the necessary statements.

**Benefits**

- Improved visibility and understanding of migration status for users.
- Empowers users to identify and resolve drifts in their system efficiently.
- Provides clear instructions and recommendations for fixing identified issues.
- Enhances user experience by centralizing update and migration information within the application.

**Related Issues**

Fixes #52265

**Screenshots**

CleanShot 2023-05-22 at 09 55 39

**Reviewer Notes**

- Please pay special attention to the accuracy of drift detection and the clarity of the instructions provided to users.
- Test the UI thoroughly with different migration scenarios to ensure its robustness.

Once the pull request is merged, users will be able to access the new UI for updates and migration information, allowing them to stay informed about their migration status and efficiently address any drifts that may occur.

## Test plan

- Thoroughly tested the new UI component with various migration scenarios.
- Passed current unit tests to ensure accurate detection of drifts and generation of deltas.
- Verified that the URL hints for fixing issues direct users to relevant resources.

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.02% (+2.58 kb) | 0.02% (+2.58 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/cj+ef/better-drift-ui/compare-report.html;tab=live_object) for a full comparison between the commits 22749c7 and 7f4755b or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: internal/database/migration/cliutil/drift.go [hunk 1]
```diff
@@ -141,15 +141,36 @@ func Drift(commandName string, factory RunnerFactory, outFactory OutputFactory,
...
+					out.WriteLine(output.Linef(output.EmojiSuccess, output.StyleSuccess, "Successfully applied autofix"))
+					out.WriteLine(output.Linef(output.EmojiInfo, output.StyleReset, "Re-checking drift"))
+				}
+
+				schemas, err := store.Describe(ctx)
+				if err != nil {
+					return err
+				}
+				schema := schemas["public"]
+				summaries = drift.CompareSchemaDescriptions(schemaName, version, Canonicalize(schema), Canonicalize(expectedSchema))
```
Comment (efritz) [inline internal/database/migration/cliutil/drift.go:173]: Is this a regression?

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -7061,13 +7061,43 @@ type UpgradeReadiness {
...
-    schemaDrift: String!
+    schemaDrift: [SchemaDriftSummary!]!
     """
     The list of deprecated-but-unfinished out-of-band migrations.
     """
     requiredOutOfBandMigrations: [OutOfBandMigration!]!
 }
 
+"""
+TODO
```
Comment (efritz) [inline cmd/frontend/graphqlbackend/schema.graphql:7072]: (_todo_)

File: internal/database/migration/cliutil/drift.go (hunk 1 above)
Comment (efritz) [reply internal/database/migration/cliutil/drift.go:173]:
```suggestion
			summaries, err = attemptAutofix(ctx, out, store, summaries, func(schema descriptions.SchemaDescription) []drift.Summary {
				return drift.CompareSchemaDescriptions(schemaName, version, canonicalize(schema), canonicalize(expectedSchema))
			})
			if err != nil {
				return err
```

Comment (Numbers88s) [reply internal/database/migration/cliutil/drift.go:173]: Oh I thought this was the correct version. It reflected as a change in this branch so I thought this was a change you added.

File: cmd/frontend/graphqlbackend/schema.graphql (hunk 2 above)
Comment (Numbers88s) [reply cmd/frontend/graphqlbackend/schema.graphql:7072]: Yup 👍, on my radar

File: internal/database/migration/cliutil/drift.go (hunk 1 above)
Comment (efritz) [reply internal/database/migration/cliutil/drift.go:173]: What if you merge a newer main in?

**Pull request merged**